Fix spc doesn't have metal attribute error

hwpang opened the pull request:
### Motivation or Problem
Fix spc doesn't have metal attribute error

### Description of Changes
Changed `r.metal` to `metal`. Also changed dummy variable `r` to `rev_react` for a meaningful name & consistency.

### Testing
Species doesn't have metal attribute error is gone after the changes

mjohnson541 commented on rmgpy/data/kinetics/family.py:
```diff
@@ -4239,14 +4239,14 @@ def fix_labels_mol(mol, root_labels):
-                            therm_spc = deepcopy(r)
+                    for rev_react in rrev.reactants:
+                        if rev_react.thermo is None:
+                            therm_spc = deepcopy(rev_react)
```
Can we avoid this deepcopy the same way we did in #2153 ?

hwpang replied:
Yeah sure!

mjohnson541 reviewed (changes requested):
Looks good! One thing.

hwpang commented:
@mjohnson541 Thanks for the review! I have added a function `get_react_thermo` to avoid deepcopy.

rwest commented on rmgpy/data/kinetics/family.py:
```diff
@@ -4049,6 +4049,23 @@ def fix_labels_mol(mol, root_labels):
                 if atm.label not in root_labels:
                     atm.label = ''
 
+        def get_react_thermo(react,rxn_react,metal):
```
From the function name and parameter names I can't guess what this is supposed to do. The meanings of `react`, `rxn_react`, and `metal` are hard to guess out of context. Maybe consider renaming them, and/or add a docstring to the function saying what it does?

hwpang replied:
@rwest Good point! Thanks

rwest reviewed (commented):
I know you were just reducing code duplication by making that function, but there's an opportunity to make it better. As it stands, I found the function quite confusing based just on its name and parameter names.

davidfarinajr commented:
Thanks for fixing this @hwpang !  I like adding the function here, but perhaps it could have just 2 arguments (`react` and `metal`) and then, since it's a get function, it could return the thermo object instead of assigning it to rxn_react.thermo attr?  So something like 
```
 if rxns[i].reactants[j].thermo is None:
      rxns[i].reactants[j].thermo = get_react_thermo(react,metal)
```

davidfarinajr commented:
Also a little bit of documentation could help, but since it's a local function inside a function, I don't think we will use it too much

hwpang commented:
@rwest @davidfarinajr Thanks for the comments! I changed the function name to `get_reactant_thermo` and added a docstring to give more explanation. I also changed the function to return the thermo object!

davidfarinajr reviewed (approved):
🙏 Thank you for fixing this @hwpang and addressing the comments !  Looks good to me!

mjohnson541 commented:
@hwpang can you rebase?

hwpang commented:
> @hwpang can you rebase?

Rebased!
